### PR TITLE
feature: make features clickable on gentk page #480

### LINKED ISSUE #480: Make traits clickable to prefill filter
It would be a great UX improvement if you guys can make the traits clickable, so it gets prefilled in the filter on the "generative project" page. For example like this screenshot:

### PR DESCRIPTION
**Pull request by alex-key**
- fix #480 

**Summary of changes:**
- make feature variant clickable on gentk page: by click user goes to the project page with selected filter selected on page load
- created `useIsMobile` and replace some duplicated functionality in some components (useWindowWidth -> useIsMobile)

**Rationale**
I did not want to make full feature block clickable, because it should be a `<button>` logically, but it would not look like button, so I decided to make only feature variant name clickable link. It redirects to `/gentk/[tokenId]/?features={...}`. On the target page url is silently updated to remove `?features=...` query, otherwise we should track all feature filters changes and reflect in the URL.
Feature filters panel will be expanded on load on desktop, but collapsed on mobile.

**Demo:**

Desktop:

feature-filter-desktop

Mobile:

feature-filter-mobile

### CODE REVIEW

File: src/components/Exploration/FiltersSubGroup.tsx [hunk 1]
```diff
@@ -27,7 +27,7 @@ export function FiltersSubGroup({
         <span>{title}</span>
         <i className={`fas fa-caret-${expanded ? "down" : "left"}`} />
       </button>
-      {expanded && <main>{children}</main>}
+      {expanded && <div>{children}</div>}
```
Comment (alexanderkorytskyi) [inline src/components/Exploration/FiltersSubGroup.tsx:30]: Not sure why `<main>` was used here, but it's a big issue in HTML. `<main>` tag should be a single one in the document.

**Comment by alex-key** [comment]
@maerzhase @GemN Could you review, please? Thx!

**Comment by maerzhase** [comment]
hej thanks for the initiative. we are currently working heavily on params. please have some patience. we will review asap

**Comment by alex-key** [comment]
> hej thanks for the initiative. we are currently working heavily on params. please have some patience. we will review asap

No pressure, Markus. I've just pinged you for the visibility, because I don't have rights to assign reviewers.

**Review by maerzhase — APPROVED** [review]
👍 thanks @louisholley for the overhaul.
and thanks @alex-key for the initial push
🐰

**Comment by alex-key** [comment]
Cool, @maerzhase & @louisholley  ! Thank you, guys, for finalizing and merge.

**Pull request merged**